File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/sample/search/OnlyListableSearchCriteria.java
package ch.ethz.sis.openbis.generic.asapi.v3.dto.sample.search;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search.AbstractCompositeSearchCriteria;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search.SearchCriteriaToStringBuilder;
import ch.systemsx.cisd.base.annotation.JsonObject;

@JsonObject("as.dto.sample.search.OnlyListableSearchCriteria")
public class OnlyListableSearchCriteria extends AbstractCompositeSearchCriteria
{

    private static final long serialVersionUID = 1L;

    @Override
    protected SearchCriteriaToStringBuilder createBuilder()
    {
        SearchCriteriaToStringBuilder builder = super.createBuilder();
        builder.setName("ONLY_LISTABLE_SAMPLES_SEARCH");
        return builder;
    }

}

File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/sample/search/OnlyListableSearchCriteria.js
/**
 * @author pkupczyk
 */
define([ "stjs", "as/dto/common/search/AbstractCompositeSearchCriteria" ], function(stjs, AbstractCompositeSearchCriteria) {
	var OnlyListableSearchCriteria = function() {
		AbstractCompositeSearchCriteria.call(this);
	};
	stjs.extend(OnlyListableSearchCriteria, AbstractCompositeSearchCriteria, [ AbstractCompositeSearchCriteria ], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.sample.search.OnlyListableSearchCriteria';
		constructor.serialVersionUID = 1;
	}, {});
	return OnlyListableSearchCriteria;
})
